### ISSUE TITLE
Siege zip encoding disable with zlib installed on mac installed via brew

### ISSUE THREAD

**Issue by counterbeing** [JoeDog/siege#94]
I've figured this out, but I think maybe it should be documented somewhere, so consider this a request for documentation.

I'm getting `Zip encoding disabled; siege requires zlib support to enable it ` warning when running siege. To fix this:

```bash
brew uninstall siege
brew install zlib
brew install --build-from-source siege
```

Thanks for the great tool!

**Comment by JoeDog** [comment]
I need to implement this better. It's compile-time dependent on zlib.
Thanks for the heads-up.

Regards,
J.

**Comment by counterbeing** [comment]
No problem, thanks for all of the work on this. Seige is super handy, and now at least the problem is google-able. 👍 

It just helped me debug some server side caching issues.

**Comment by JoeDog** [comment]
I'm glad you like it.

J.

**Comment by anotheri** [comment]
unfortunately that fix w/ `--build-from-source` doesn't work for me:
```
siege 4.0.2
zlib 1.2.8
macOS 10.12.2
```
any other suggestions how to fix it?

**Comment by JoeDog** [comment]
How is it failing?

**Comment by anotheri** [comment]
@JoeDog it shows output like this and no results:
```
> siege -q -c10 -t10s "http://localhost:3000/"
[alert] Zip encoding disabled; siege requires zlib support to enable it

Lifting the server siege...%
```

**Comment by anotheri** [comment]
@JoeDog or without `-q` it looks like this:

```
> siege -c10 -t10s http://localhost:3000
[alert] Zip encoding disabled; siege requires zlib support to enable it
** SIEGE 4.0.2
** Preparing 10 concurrent users for battle.
The server is now under siege...
HTTP/1.1 200     1.60 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     1.60 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     1.60 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     1.60 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     1.60 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     1.61 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     1.61 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     1.61 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     9.14 secs:  227220 bytes ==> GET  /
HTTP/1.1 200     9.14 secs:  227220 bytes ==> GET  /

Lifting the server siege...
```

and it is hanging, no results as well.

**Comment by JoeDog** [comment]
This MacOS, right?

That hanging with no result seems to occur a lot on MacOS and I don't have
a Mac to test it on. (I suspect it has something to do with signalling
there. You can get around that problem by scheduling repetitions instead of
time, i.e., siege -c10 -r10 http://localhost:3000/

For the build portion, I need to see what happens when you run ./configure

(It must not be finding your library; where is zlib.h installed)

**Comment by ronpeled** [comment]
Just to keep this thread going. Both of these issues are still occurring on MacOS 10.12.3 with latest brew installed. The two issues:

1. Zip encoding disabled... even if I install zlib and run the --build-from-source command.
2. if I run a simple benchmarking command with 60 seconds, it hangs and never finishes.

Thanks!

**Comment by norbour** [comment]
confused by the same problem

**Comment by jonthemiller** [comment]
Same issue on macOS 10.12.2. I have the `zlib` and `lzlib` packages also installed with Homebrew. here's the portion of the output from `brew install --build-from-source -v siege` pertaining to zlib:

```
checking for zlib support... yes
checking /include/zlib.h usability... no
checking /include/zlib.h presence... no
checking for /include/zlib.h... no
checking /usr/include/zlib.h usability... no
checking /usr/include/zlib.h presence... no
checking for /usr/include/zlib.h... no
checking /usr/local/include/zlib.h usability... no
checking /usr/local/include/zlib.h presence... no
checking for /usr/local/include/zlib.h... no
checking /usr/local/ssl/include/zlib.h usability... no
checking /usr/local/ssl/include/zlib.h presence... no
checking for /usr/local/ssl/include/zlib.h... no
checking /usr/pkg/include/zlib.h usability... no
checking /usr/pkg/include/zlib.h presence... no
checking for /usr/pkg/include/zlib.h... no
checking /usr/lib/zlib/include/zlib.h usability... no
checking /usr/lib/zlib/include/zlib.h presence... no
checking for /usr/lib/zlib/include/zlib.h... no
checking /usr/include/zlib/include/zlib.h usability... no
checking /usr/include/zlib/include/zlib.h presence... no
checking for /usr/include/zlib/include/zlib.h... no
checking /usr/include/include/zlib.h usability... no
checking /usr/include/include/zlib.h presence... no
checking for /usr/include/include/zlib.h... no

**Comment by pasha** [comment]
I had similar issues on macOS 10.12.4, where server never finish it's run.  After following reinstall instructions in original post, i noticed some warnings about `sysctl net.inet.tcp.msl:15000`, I updated settings to 1 sec `sudo sysctl net.inet.tcp.msl:1000`.

**Comment by fukuball** [comment]
same as @anotheri

**Comment by pwaldo2** [comment]
I managed to stop getting the zlib warning with a combination of xcode-select --install and updating the TIME_WAIT to 1 second, but I still get a 400 error on my JSON POST test. Sending the JSON over with Postman works just fine.

**Comment by jonasanso** [comment]
I followed the instructions and worked nicely for me.
Only detail to add I had an old zlib version and I ran `brew upgrade zlib`.

Thanks Joe for this great tool.

Note to @pwaldo2 : I am executing a json POST without warnings.
```
siege -c5 -t1M -H 'Content-Type: application/json' 'http://localhost:9200/user/_search POST { "query": { "query_string": {"query": "daniel"}}}'
```

**Comment by vikas027** [comment]
Didn't worked for me too on OSX `10.14`

**Comment by emanserav** [comment]
encountering the same problem on OS Mojave , Version 10.14.2 (18C54)

**Comment by jasonmccreary** [comment]
Was encountering the same issues as @jonthemiller noted. Found a potential solution on [another projects thread](https://github.com/pyenv/pyenv/issues/1219).

**tl;dr;** Xcode command line tools not longer includes the header files. To install them run:

```sh
sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /```

After doing so I rebuild from source with the following output:

```txt
checking for zlib support... yes
checking /include/zlib.h usability... no
checking /include/zlib.h presence... no
checking for /include/zlib.h... no
checking /usr/include/zlib.h usability... yes
checking /usr/include/zlib.h presence... yes
checking for /usr/include/zlib.h... yes
checking for ZLIB version... checking for inline... inline
```

**Comment by anapsix** [comment]
I've had to install zlib via `brew install zlib`, and edit siege recipe with `brew edit siege`
updated to following
```
system "./configure", "--disable-dependency-tracking",¬
                      "--prefix=#{prefix}",¬
                      "--mandir=#{man}",¬
                      "--localstatedir=#{var}",¬
                      "--with-ssl=#{Formula["openssl"].opt_prefix}",¬
                      "--with-zlib=#{Formula["zlib"].opt_prefix}"¬
```
Works as expected after this modification, and installation with `brew install siege --build-from-source`

**Comment by anapsix** [comment]
confirmed https://github.com/Homebrew/homebrew-core/pull/37342 solves zlib related issue

**Comment by mistydemeo** [comment]
I've opened #152 to improve the configure check so it can find macOS's builtin zlib.
